internal/attestation/aws/snp/validator.go

```diff
@@ -9,99 +9,216 @@ package snp
 import (
 	"context"
 	"crypto"
+	"crypto/sha512"
+	"crypto/x509"
 	"encoding/json"
 	"fmt"
 
-	awsConfig "github.com/aws/aws-sdk-go-v2/config"
-	"github.com/aws/aws-sdk-go-v2/feature/ec2/imds"
-	"github.com/aws/aws-sdk-go-v2/service/ec2"
 	"github.com/edgelesssys/constellation/v2/internal/attestation"
+	"github.com/edgelesssys/constellation/v2/internal/attestation/snp"
 	"github.com/edgelesssys/constellation/v2/internal/attestation/variant"
 	"github.com/edgelesssys/constellation/v2/internal/attestation/vtpm"
 	"github.com/edgelesssys/constellation/v2/internal/config"
+	"github.com/google/go-sev-guest/abi"
+	"github.com/google/go-sev-guest/kds"
+	"github.com/google/go-sev-guest/proto/sevsnp"
+	"github.com/google/go-sev-guest/validate"
+	"github.com/google/go-sev-guest/verify"
+	"github.com/google/go-sev-guest/verify/trust"
 	"github.com/google/go-tpm-tools/proto/attest"
 	"github.com/google/go-tpm/legacy/tpm2"
 )
 
 // Validator for AWS TPM attestation.
 type Validator struct {
+	// Embed variant to identify the Validator using varaint.OID().
 	variant.AWSSEVSNP
+	// Embed validator to implement Validate method for aTLS handshake.
 	*vtpm.Validator
-	getDescribeClient func(context.Context, string) (awsMetadataAPI, error)
+	// AMD root key. Root of trust for the ASK used during report validation.
+	ark *x509.Certificate
+	// AMD signing key. Intermediate certificate that connects ARK and report signing key (leaf certificate).
+	ask *x509.Certificate
+	// cfg contains version numbers required for the SNP report validation.
+	cfg *config.AWSSEVSNP
```

Comment:
why do we duplicate ark and ask here if we also store the config?

Reply:
that does make no sense, right.